stdlib/src/builtin/string.mojo

```diff
@@ -1549,3 +1537,173 @@ fn _calc_initial_buffer_size[type: DType](n0: Scalar[type]) -> Int:
             )
 
     return 128 + 1  # Add 1 for the terminator
+
+
+@value
+struct _InlineBytesList[capacity: Int](Sized, CollectionElement):
```

Comment:
**Thought** I'm not convinced this is a better solution than making `InlineFixedVector` use an `InlineVector` instead of `StaticTuple`, but that's also less trivial an update.

Reply:
That's a great idea! I didn't know InlineFixedVector  existed actually. There is not much code to change and can be a good way to split this PR into multiple ones. Definitly in my TODO list.

Reply:
FYI `InlineFixedVector` is going to switch over to `AnyType` or `CollectionElement` after we fix the compiler bug(s) blocking https://github.com/modularml/mojo/pull/2377